internal/constraints/constraints.go

```diff
@@ -1,21 +1,91 @@
 package constraints
 
 import (
+	"bytes"
 	"fmt"
 	"os"
 	"regexp"
 	"sort"
 	"strconv"
 	"strings"
+	"text/template"
+
+	"github.com/ActiveState/sysinfo"
+	"github.com/thoas/go-funk"
 
 	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
 	"github.com/ActiveState/cli/pkg/projectfile"
-	"github.com/ActiveState/sysinfo"
 )
 
 // For testing.
 var osOverride, osVersionOverride, archOverride, libcOverride, compilerOverride string
 
+type Conditional struct {
+	params map[string]interface{}
+	funcs  template.FuncMap
+}
+
+func NewConditional() *Conditional {
+	c := &Conditional{map[string]interface{}{}, map[string]interface{}{}}
+
+	c.RegisterFunc("Contains", funk.Contains)
+	c.RegisterFunc("HasPrefix", strings.HasPrefix)
+	c.RegisterFunc("HasSuffix", strings.HasSuffix)
+	c.RegisterFunc("MatchRx", func(rxv, v string) bool {
+		rx, err := regexp.Compile(rxv)
+		if err != nil {
+			logging.Warning("Invalid Regex: %s, error: %v", rxv, err)
+			return false
+		}
+		return rx.Match([]byte(v))
+	})
+
+	return c
+}
+
+func NewPrimeConditional(pjOwner, pjName, pjNamespace, subshellName string) *Conditional {
+	c := NewConditional()
+	c.RegisterParam("Project", map[string]string{
+		"Namespace": pjNamespace,
+		"Name":      pjName,
+		"Owner":     pjOwner,
+	})
+	osVersion, err := sysinfo.OSVersion()
+	if err != nil {
+		logging.Error("Could not detect OSVersion: %v", err)
+	}
+	c.RegisterParam("OS", map[string]interface{}{
+		"Name":         sysinfo.OS().String(),
+		"Version":      osVersion,
+		"Architecture": sysinfo.Architecture().String(),
+	})
+	c.RegisterParam("Shell", subshellName)
+
+	return c
+}
+
+func (c *Conditional) RegisterFunc(name string, value interface{}) {
+	c.funcs[name] = value
+}
+
+func (c *Conditional) RegisterParam(name string, value interface{}) {
+	c.params[name] = value
+}
+
+func (c *Conditional) Eval(conditional string) (bool, error) {
+	tpl, err := template.New("letter").Funcs(c.funcs).Parse(fmt.Sprintf(`{{if %s}}1{{end}}`, conditional))
```

Comment:
:+1: for piggy-backing on the go template parser to implement all the complex parsing in one line.